crates/compiler/src/obj_gen/mod.rs

```diff
@@ -2660,6 +2686,307 @@ impl ObjectGenerator {
         }
     }
 
+    /// Generates the FLO code that corresponds to an LLVM
+    /// [indirectbr](https://llvm.org/docs/LangRef.html#indirectbr-instruction)
+    /// instruction.
+    ///
+    /// # Errors
+    ///
+    /// - [`Error`] if the correct FLO representation of the instruction cannot
+    ///   be generated.
+    ///
+    /// # Panics
+    ///
+    /// - If the provided instruction is _not_ an `indirectbr`.
+    pub fn generate_indirect_br(
+        &self,
+        instruction: InstructionValue,
+        bb: &mut BlockBuilder,
+        func_ctx: &mut FunctionContext,
+    ) -> Result<()> {
+        assert_correct_opcode(&instruction, InstructionOpcode::IndirectBr);
+
+        // We have to deal with the blockaddress constants as the only valid source of
+        // the address operand. We currently do not handle these properly.
+
+        // 1. We get a `get_block_address` statement that takes a block id and puts it
+        //    in an SSA variable. This is non-local, but we assume it is within the same
+        //    TU, panicking if not. GetBlockAddress(var, block_id, diagnostics, loc)  is
+        //    a new stmt + struct pair.
+        // 2. Then the compiler can generate the comparisons.
+        // 3. Uses these in a match.
+
+        // Let us start by getting the operands to the indirectbr instruction. These
+        // include both the pointer operand, and the potential target blocks, so we
+        // cannot perform any global extractions.
+        let operands = instruction.get_operands().collect_vec();
+
+        // The first operand is the pointer that determines how to dispatch.
+        let pointer = util::extract_value_operand(
+            *operands.first().ok_or_else(|| {
+                let actual_arg_count = instruction.get_num_operands();
+                Error::MalformedLLVM(format!(
+                    "IndirectBr instruction had {actual_arg_count} operands where at least 1 was \
+                     expected"
+                ))
+            })?,
+            InstructionOpcode::IndirectBr,
+        )?;
+        let pointer_id = util::get_var_or_const(&pointer, bb, func_ctx)?;
+
+        // The remaining ones are the potential block labels in the function, which we
+        // can instantly turn into variables for comparison against. From the LLVM IR
```

Comment:
Generally, I prefer citations to "from the langref", e.g. `[LLVM Langref <section header>]` or a permalink,